### ISSUE TITLE
Optional PO number

### ISSUE THREAD

**Issue by springimport** [magento/magento2#6585]
### Steps to reproduce
1. Add items to cart, go to checkout and select "Purchase Order".
   ![111](https://cloud.githubusercontent.com/assets/18581994/18447920/a94edfa8-78f6-11e6-83a1-4c271454fb81.PNG)
2. Use REST API `[POST] /V1/carts/mine/payment-information`

``` json
{
  "paymentMethod": {
      "method": "purchaseorder"
   }
}
```
### Expected result

Error: PO number is required.
### Actual result

``` json
"90"
```

(number of new order)
### Additional info

[API documentation](http://devdocs.magento.com/swagger/index.html) checkoutPaymentInformationManagementV1

```
quote-data-payment-interface {
  po_number (string, optional): Purchase order number ,
  ...
}
```

| Q | A |
| --- | --- |
| Magento version | 2.1.0 with sample data |
| PHP version | 7 x64 |
| Operating system | Windows 10 x64 |

**Comment by vityakopin** [comment]
Magento does not support your operating system. Please confirm that the issue reproduces  on some supported system as well. Please see [Magento technology stack requirements](http://devdocs.magento.com/guides/v2.1/install-gde/system-requirements-tech.html) for more information.

**Comment by smoskaluk** [comment]
Internal ticket created MAGETWO-70558

**Comment by magento-engcom-team** [comment]
@springimport, thank you for your report.
We've created internal ticket(s) MAGETWO-70558 to track progress on the issue.

**Comment by matin73** [comment]
Bug still exists in Magento 2.2.4 - will this bug be fixed soon?

**Comment by resolvr** [comment]
no never

**Comment by resolvr** [comment]
@matin73, i don't think so..

**Comment by miguelbalparda** [comment]
There is a fix for this in #14393. Please give it a try and report back @matin73.

**Comment by sidolov** [comment]
Hi @springimport. Thank you for your report. 
The issue has been fixed in magento/magento2#14393 by @centerax in 2.2-develop branch
Related commit(s):
 - [654f350bf6ecd61fd59f6c79949db869661bb17e](https://github.com/magento/magento2/commit/654f350bf6ecd61fd59f6c79949db869661bb17e)
 - [f5adaa3dd75b26643cbe9ac260dcab37e59d6d30](https://github.com/magento/magento2/commit/f5adaa3dd75b26643cbe9ac260dcab37e59d6d30)

The fix will be available with the upcoming 2.2.7 release.

**Comment by magento-engcom-team** [comment]
Hi @springimport. Thank you for your report. 
The issue has been fixed in magento/magento2#17670 by @mage2pratik in 2.3-develop branch
Related commit(s):
 - [fa3ed73176359bb03917a7f487d00410f250bb7c](https://github.com/magento/magento2/commit/fa3ed73176359bb03917a7f487d00410f250bb7c)
 - [55753f155a4d6580a0b78e84a2d79ee9b276ef4e](https://github.com/magento/magento2/commit/55753f155a4d6580a0b78e84a2d79ee9b276ef4e)
 - [1438d2583756eea24a2895a45bacf100be7b9587](https://github.com/magento/magento2/commit/1438d2583756eea24a2895a45bacf100be7b9587)

The fix will be available with the upcoming 2.3.0 release.
